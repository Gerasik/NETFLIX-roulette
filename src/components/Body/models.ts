import * as Models from 'models';
import * as BodyModels from 'containers/Body/models';

export interface BodyProps {
  setData: BodyModels.ActionCreator.SetData;
  moviesData: Models.MoviesDataMap;
}
