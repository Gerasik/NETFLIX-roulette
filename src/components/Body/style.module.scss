.movie-list {
  list-style: none;
  margin: 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 1em 3em 2em;
  background-color: $primaryColor;
  .movie-item {
    width: 30%;
  }
}
