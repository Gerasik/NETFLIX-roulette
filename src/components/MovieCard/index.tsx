import React from 'react';
import styles from './style.module.scss';

import { MovieCardData } from '../../models';

type MovieCardProps = {
  data: MovieCardData;
};

export default function MovieCard(props: MovieCardProps) {
  const { data } = props;
  const { title, genres, poster_path, release_date } = data;
  return (
    <li className={styles.item}>
      <img src={poster_path} alt={title} className={styles.image} />
      <div className={styles.container}>
        <span className={styles.title}>{title}</span>
        <span className={styles.date}>{new Date(release_date).getFullYear()}</span>
      </div>
      {genres.map((i, p) => {
        const keyId = p + 1;
        return (
          <span key={keyId} className={styles.gengre}>
            {i}
          </span>
        );
      })}
    </li>
  );
}
