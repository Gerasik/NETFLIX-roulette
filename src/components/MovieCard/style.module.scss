@import '../../styles/colors';

$apacityColor: rgba(
  $color: $font-color,
  $alpha: 0.7,
);

$apacityColorSecond: rgba(
  $color: $font-color,
  $alpha: 0.4,
);

.item {
  color: $apacityColor;
  .image {
    width: 100%;
    margin-top: 2em;
  }
  .title {
    font-weight: bold;
  }
  .date {
    padding: 0.5em 1.3em;
    border: 0.05em solid $apacityColorSecond;
    border-radius: 0.8em;
  }
  .container {
    display: flex;
    justify-content: space-between;
    margin-top: 0.8em;
    align-content: center;
  }
  .gengre {
    color: $apacityColorSecond;
    padding-right: 0.4em;
  }
}
