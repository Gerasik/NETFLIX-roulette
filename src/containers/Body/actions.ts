import ActionType from './constants';
import { Action } from '../../models';

export const setData: Action = data => {
  return { type: ActionType.SET_DATA, payload: data };
};

export const increment: Action = data => {
  return { type: ActionType.SET_DATA, payload: data };
};
