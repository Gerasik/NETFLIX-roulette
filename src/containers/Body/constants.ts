const ActionType = {
  SET_DATA: 'SET-DATA',
};

export default ActionType;
