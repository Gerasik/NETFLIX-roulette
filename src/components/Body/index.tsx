import React, { useEffect, FunctionComponent } from 'react';
import classNames from 'classnames';

import { Action } from 'containers/Search/models';
import MovieCard from 'components/MovieCard';
import styles from './style.module.scss';
import { BodyProps } from './models';

const Body: FunctionComponent<BodyProps> = ({ setData, moviesData, changeSortBy }) => {
  const { data, total, sortBy } = moviesData;

  useEffect(() => {
    fetch('https://reactjs-cdp.herokuapp.com/movies?limit=6')
      .then(response => response.json())
      .then(response => {
        setData(response);
      });
  }, [setData]);

  return (
    <div className={styles['movie-container']}>
      <header className={styles['movie-header']}>
        <span className={styles['movie-count']}>{total ? `${total} movie found` : ''}</span>
        <div className={styles['movie-buttons']}>
          <span>sort by</span>
          <button
            type="button"
            className={classNames(styles['sort-release'], {
              [styles.active]: sortBy === 'release_date',
            })}
            onClick={(): Action.CHANGE_SORT_BY => changeSortBy('release_date')}
          >
            release date
          </button>
          <button
            type="button"
            className={classNames(styles['sort-rating'], {
              [styles.active]: sortBy === 'vote_average',
            })}
            onClick={(): Action.CHANGE_SORT_BY => changeSortBy('vote_average')}
          >
            rating
          </button>
        </div>
      </header>
      <ul className={styles['movie-list']}>
        {data.map(item => (
          <li key={item.get('id')} className={styles['movie-item']}>
            <MovieCard data={item} />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Body;
