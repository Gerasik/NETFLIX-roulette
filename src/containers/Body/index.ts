import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import { setData } from './actions';
import { State } from './models';
import Body from '../../components/Body';

const bodySelector = createSelector(
  [(state: { toJS: () => { bodyReducer: State } }): any => state.toJS()],
  bodyReducer => bodyReducer
);

const mapStateToProps = (state: { toJS: () => { bodyReducer: State } }): any => ({
  bodyReducer: bodySelector(state),
});

const mapDispatchToProps = {
  setData,
};

export default connect(mapStateToProps, mapDispatchToProps)(Body);
