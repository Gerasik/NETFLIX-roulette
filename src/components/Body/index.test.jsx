/* eslint-disable @typescript-eslint/camelcase */
import configureStore from 'react-redux';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Immutable from 'immutable';

import MovieCard from 'components/MovieCard';
import Body from './index';

const testMoviesData = Immutable.fromJS([
  {
    id: 1,
    title: 'first movie',
    tagline: 'test',
    vote_average: 55,
    vote_count: 66,
    release_date: '08081992',
    poster_path: '/movie_url/image.jpg',
    overview: 'ove',
    budget: 54545,
    revenue: 5454,
    runtime: 5454,
    genres: ['drama', 'sitcom', 'drama'],
  },
  {
    id: 2,
    title: 'second movie',
    tagline: 'test',
    vote_average: 55,
    vote_count: 66,
    release_date: '08081992',
    poster_path: '/movie_url/image2.jpg',
    overview: 'ove',
    budget: 54545,
    revenue: 5454,
    runtime: 5454,
    genres: ['drama'],
  },
]);

let returnData;

function testSetData(response) {
  returnData = response;
}

describe('<Body />', () => {
  const wrapper = shallow(<Body moviesData={testMoviesData} />);

  it('should render with test data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be two MovieCard component', () => {
    expect(wrapper.find(MovieCard).length).toBe(2);
  });

  it('right props first MovieCard component', () => {
    expect(wrapper.find(MovieCard).get(0).props.data).toBe(testMoviesData.get(0));
  });
});
