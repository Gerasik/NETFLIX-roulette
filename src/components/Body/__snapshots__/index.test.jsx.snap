// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body /> should render with test data 1`] = `
<ul
  className="movie-list"
>
  <li
    className="movie-item"
    key="1"
  >
    <MovieCard
      data={
        Immutable.Map {
          "budget": 54545,
          "release_date": "08081992",
          "vote_average": 55,
          "tagline": "test",
          "genres": Immutable.List [
            "drama",
            "sitcom",
            "drama",
          ],
          "poster_path": "/movie_url/image.jpg",
          "title": "first movie",
          "runtime": 5454,
          "overview": "ove",
          "vote_count": 66,
          "id": 1,
          "revenue": 5454,
        }
      }
    />
  </li>
  <li
    className="movie-item"
    key="2"
  >
    <MovieCard
      data={
        Immutable.Map {
          "budget": 54545,
          "release_date": "08081992",
          "vote_average": 55,
          "tagline": "test",
          "genres": Immutable.List [
            "drama",
          ],
          "poster_path": "/movie_url/image2.jpg",
          "title": "second movie",
          "runtime": 5454,
          "overview": "ove",
          "vote_count": 66,
          "id": 2,
          "revenue": 5454,
        }
      }
    />
  </li>
</ul>
`;
